### ISSUE TITLE
Error "Object literal may only specify known properties" is not happening in all cases with intersection types

### ISSUE THREAD

**Issue by asfernandes** [Microsoft/TypeScript#18075]
**TypeScript Version:**  2.4.2

```ts
export type MyType =
	{
		id: number;
	}
	&
	{
		name: string;
	}
	&
	{
		photo:
		{
			id: number;
		}
		&
		{
			url: string;
		}
	}


export let obj: MyType;

export const photo: typeof obj.photo = {
	id: 1,
	url: '',
	xyz: 1	// Great! This causes an error!
};
	
export const myInstance: MyType = {
	id: 1,
	name: '',
	photo: {
		id: 1,
		url: '',
		xyz: 2	// Bug? This does not cause an error!!!
	}
};
```

**Expected behavior:**

I expect all attributions to `xyz` to give an error.

**Actual behavior:**

TypeScript is accepting one of the attributions to `xyz` (with value 2).

**Comment by asfernandes** [comment]
Workaround:

```ts
export type Wrapper<T> = {
	[P in keyof T]: T[P]
}

export type MyType = Wrapper<
	{
		id: number;
		name: string;
	}
	&
	{
		photo: PhotoType
	}
>
```

**Comment by mhegazy** [comment]
Excess property checks are disabled on intersection types. wondering if this pattern is just a test, or part of a real world code?

**Comment by asfernandes** [comment]
This is the smallest test case I was able to demonstrate the problem I found in my real code.

**Comment by ekilah** [comment]
@mhegazy for what it's worth, and maybe for a bump, I'm also experiencing this bug in real-world code.

**Comment by ekilah** [comment]
for others getting here, my workaround was to use an `interface X extends Y {...}` rather than `type X = Y & {...}`, but I've run into other cases elsewhere where interfaces are not the best choice for other reasons... just another thing to remember about the difference between them, I guess.

Can we get more information about what "Excess property checks are disabled on intersection types" actually means @mhegazy ?

**Comment by PavelPZ** [comment]
Is any chance to provide more specific information (other than milestones: **Future**) about planned solution? 

In https://github.com/Microsoft/TypeScript/issues/27448#event-2024921711 is much more simple example of this bug. 

Thanks a lot and congratulation to great Typescript product!!!
